host/greatfet/commands/greatfet_logic.py

```diff
@@ -38,20 +40,43 @@ def main():
             print("No GreatFET board found!", file=sys.stderr)
         sys.exit(errno.ENODEV)
 
-    device.apis.logic_analyzer.start()
+    # pulseview metadata
+    metadata_str = ("[device 1]\n"
+                    "capturefile=%s\n" 
+                    "total probes=8\n"
+                    "samplerate=17 MHz\n"
+                    "total analog=0\n"
+                    "probe1=SGPIO0\n"
+                    "probe2=SGPIO1\n"
+                    "probe3=SGPIO2\n"
+                    "probe4=SGPIO3\n"
+                    "probe5=SGPIO4\n"
+                    "probe6=SGPIO5\n"
+                    "probe7=SGPIO6\n"
+                    "probe8=SGPIO7\n"
+                    "unitsize=1\n") % args.filename 
 
+    device.apis.logic_analyzer.start()
     time.sleep(1)
-
-    f = open(args.filename, 'wb')
+    fd, path = tempfile.mkstemp()
+    print("Press Ctrl+C to stop")
     try:
-        while True:
-            d = device.comms.device.read(0x81, 16384, 1000)
-            f.write(d)
-    except KeyboardInterrupt:
-        pass
+        with os.fdopen(fd, "wb") as tmp:
+            try:
+                while True:
+                    d = device.comms.device.read(0x81, 16384, 1000)
+                    tmp.write(d)
+            except KeyboardInterrupt:
+                pass
+            # pulseview compatible .sr archive
+            with ZipFile("session.sr", "w") as zip:
```

Comment:
Should this use the `filename` argument instead of hardcoding session.sr?

Reply:
The binary data file is what's being named with filename, should this as well?

Reply:
I think so, it feels more consistent that the filename a user provides on the command line matches the one that they're looking for in the pulseview find file dialog box